test/widgets/compose_box_test.dart

```diff
@@ -668,4 +681,81 @@ void main() {
       });
     });
   });
+
+  group('ComposeBox content input scaling', () {
+    const verticalPadding = 8;
+    final stream = eg.stream();
+    final narrow = TopicNarrow(stream.streamId, 'foo');
+
+    Future<void> checkContentInputMaxHeight(WidgetTester tester, {
+      required double maxHeight,
+      required int maxVisibleLines,
+    }) async {
+      final finder = find.byType(ComposeBox);
+
+      double? height;
+      // The content input contains two lines at minimum.
+      String content = 'foo\nfoo';
+
+      // On top of the existing two lines, add exactly the number of lines
+      // needed such that the last line will not visible in the content input
+      // after reaching the max height.
+      for (int i = 0; i < maxVisibleLines - 2 + 1; i++) {
+        // Add one line at a time,
+        // until the content input reaches its max height.
+        await tester.enterText(finder, content);
```

Comment:
nit: just inline `finder`; the name "finder" isn't very informative about what this line is doing

Or use `contentInputFinder`? That'd match the `getRect` call just below, and its name is clear.